deluge/core/torrentmanager.py

```diff
@@ -795,6 +796,22 @@ def load_resume_data_file(self):
             return resume_data
 
     def save_resume_data_file(self):
+        """
+        Save resume data to file in a separate thread to avoid blocking main thread.
+
+        If a save file task is already running, this call is queued.
```

Comment:
I knew this seemed familiar, could we not use the same implementation as we did for [`save_state`](https://github.com/bendikro/deluge/blob/develop-save-fastresume-separate-thread/deluge/core/torrentmanager.py#L662)